pyvista/plotting/widgets.py

```diff
@@ -622,6 +622,107 @@ def callback(normal, loc):
 
         return self.add_mesh(plane_clipped_mesh, **kwargs)
 
+    def add_volume_clipper(
```

Comment:
This name too closely follows the convention of the other methods in this module when the behavior/inputs are very different. 

IMO, this method should be changed to `add_volume_clip_plane()` and take `UniformGrid` or `RectilinearGrid` as input and plot those from here.

Reply:
How about both? You can either add a clip plane on an existing `volume.Volume` actor, or add a dataset.

Advantage of this approach is you can seamlessly add multiple clip planes to a single actor. For example:

```py
pl = pv.Plotter(theme=pv.themes.DarkTheme())
vol = pl.add_volume(grid, opacity='linear', cmap='magma')
pl.add_volume_clipper(vol, normal='-x', implicit=True)
pl.add_volume_clipper(vol, normal='-y', implicit=True)
pl.show_axes()
pl.show()
```

![tmp](https://user-images.githubusercontent.com/11981631/212517515-72cd38ec-d0e9-4837-959e-b76ab22c7834.png)